Branch-SDK/src/io/branch/referral/BranchLinkData.java

```diff
@@ -397,4 +397,30 @@ public int hashCode() {
         return result;
     }
 
+    /**
+     * Creates the Json object with link params and link properties.
+     *
+     * @return {@link JSONObject} with link params and link properties
+     */
+    public JSONObject getLinkDataJsonObject() {
+        JSONObject linkDataJson = null;
+        try {
+            if (has(Defines.Jsonkey.Data.getKey())) {
+                linkDataJson = new JSONObject(getString(Defines.Jsonkey.Data.getKey()));
+
+                linkDataJson.put("~" + Defines.LinkParam.Channel.getKey(), getChannel());
```

Comment:
Ahh good catch . It cannot be null with BUO but yes if we implement the share sheet without BUO. Will Add the safe check